Improve Performance of XDR IR Integration by Optimizing Field Fetching (#37664)
demisto/content: Packs/CortexXDR/ReleaseNotes/6_2_4.md

#### Integrations

##### Palo Alto Networks Cortex XDR - Investigation and Response

> - **Breaking Changes**: Set **Remove legacy incident fields** to default true- which means that the incident data before going through the mapper is in the following format word1_word2 (and not word1word2) for example: alert_id and not alertid, and that in the context data keys in alerts, file_artifacts and network_artifacts will appear in the word1_word2 format if are originated from the API request.**Important None:** We will soon not support this parameter, meaning we will remove all fields of format word1word2(eg. alertid) from the alerts, **Make sure to check and update the custom content in order it to work with the word1_word2 (eg. alert_id) only.**

Reviewer: ```suggestion
- **Breaking Changes**: Set **Remove legacy incident fields** to default true. This means that the incident data before going through the mapper is in the following format word1_word2 (and not word1word2) for example: alert_id and not alertid, and that in the context data keys in alerts, file_artifacts and network_artifacts will appear in the word1_word2 format if originated from the API request.**Important None:** Soon we will no longer support this parameter, meaning we will remove all fields of format word1word2 (e.g., alertid) from the alerts, **Make sure to check and update the custom content in order to work with the word1_word2 (eg. alert_id) only.**
```

> - **Breaking Changes**: Deleted the **Minimize Incident Information** parameter. All incidents pulled to XSOAR will include the file artifacts and networks artifacts.

Reviewer: ```suggestion
- **Breaking Changes**: Deleted the *Minimize Incident Information* parameter. All incidents pulled to Cortex XSOAR will include the file artifacts and networks artifacts.
```

> - **Breaking Changes**: Added the **Minimize Alert Information** parameter. Default Value: null_values. This parameter will filter out the specified fields from the XDR alerts. If null_values is specified, will remove all None values from the alerts data.

Reviewer: ```suggestion
- **Breaking Changes**: Added the *Minimize Alert Information* parameter. Default value: null_values. This parameter will filter out the specified fields from the Cortex XDR alerts. If null_values is specified, will remove all empty values from the alerts data.
```

> - **Breaking Changes**: Added the **Remove Additional Data** parameter. Default Value: True. This parameter will remove the Additional Data field (additionaldata in the context data) in cases which the incident type is XDR incident Lite or Malware Investigation and Response. **Important None:** We will soon not support this parameter, meaning we will remove the Additional Data field (additionaldata in the context data) from the relevant incidents.

Reviewer: ```suggestion
- **Breaking Changes**: Added the *Remove Additional Data* parameter. Default value: True. This parameter will remove the Additional Data field (additionaldata in the context data) in cases which the incident type is XDR incident Lite or Malware Investigation and Response. **Important None:** Soon we will no longer support this parameter, meaning we will remove the Additional Data field (additionaldata in the context data) from the relevant incidents.
```

#### Mappers

##### XDR - Incoming Mapper

> - Updated the Mapping for **Cortex XDR - Lite** incident type to map only unmapped field to lables.

Reviewer: ```suggestion
- Updated the Mapping for **Cortex XDR - Lite** incident type to map only unmapped fields to labels.
```

- Updated the Mapping for **Cortex XDR - Lite** incident type:
  - The following fields will use word1_word2 format as the source of the field:
    - Additional Data uses additional_data
    - Agent Version uses agent_version
    - Cloud Provider uses cloud_provider
    - Device Name uses host_name
    - Parent Process Name uses alerts.causality_actor_process_image_name
  - The following fields will now populate only unique values:
    - External Category
    - Parent Process Name
    - Parent CMD line
    - Parent Process CMD
    - Parent Process MD5
    - Parent Process Name
    - Parent Process SHA256
    - Process CMD
    - Process MD5
    - Process SHA256
    - XDR Alert Category
    - XDR MITRE Tactics
    - XDR MITRE Techniques
    - Agent Version
    - Agents ID
    - CMD line
    - Device Internal IPs
    - External Category Name
    - File Names
    - File Paths
    - Parent CMD line

> - Updated the Mapping for **Cortex XDR Incident** incident type:

Reviewer: ```suggestion
- Updated the mapping for **Cortex XDR Incident** incident type:
```

  - The following fields will now populate only unique values:
    - XDR Alert Category
    - XDR Alert Name
    - XDR MITRE Tactics
    - XDR MITRE Techniques

> - Updated the Mapping for **Malware Investigation and Response** incident type:

Reviewer: ```suggestion
- Updated the mapping for **Malware Investigation and Response** incident type:
```
